be/src/exec/pipeline/scan/olap_scan_context.cpp

```diff
@@ -0,0 +1,75 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "exec/pipeline/scan/olap_scan_context.h"
+
+#include "exec/vectorized/olap_scan_node.h"
+#include "exprs/vectorized/runtime_filter_bank.h"
+
+namespace starrocks::pipeline {
+
+using namespace vectorized;
+
+Status OlapScanContext::prepare(RuntimeState* state) {
+    const auto& conjunct_ctxs = _scan_node->conjunct_ctxs();
+    RETURN_IF_ERROR(Expr::prepare(conjunct_ctxs, state));
+    RETURN_IF_ERROR(Expr::open(conjunct_ctxs, state));
+
+    return Status::OK();
+}
+
+void OlapScanContext::close(RuntimeState* state) {
+    const auto& conjunct_ctxs = _scan_node->conjunct_ctxs();
+    Expr::close(conjunct_ctxs, state);
+}
+
+Status OlapScanContext::parse_conjuncts(RuntimeState* state, const std::vector<ExprContext*>& runtime_in_filters,
+                                        RuntimeFilterProbeCollector* runtime_bloom_filters) {
+    const TOlapScanNode& thrift_olap_scan_node = _scan_node->thrift_olap_scan_node();
+    const TupleDescriptor* tuple_desc = state->desc_tbl().get_tuple_descriptor(thrift_olap_scan_node.tuple_id);
+
+    // Get _conjunct_ctxs.
+    _conjunct_ctxs = _scan_node->conjunct_ctxs();
+    _conjunct_ctxs.insert(_conjunct_ctxs.end(), runtime_in_filters.begin(), runtime_in_filters.end());
+
+    // eval_const_conjuncts.
+    Status status;
+    RETURN_IF_ERROR(vectorized::OlapScanConjunctsManager::eval_const_conjuncts(_conjunct_ctxs, &status));
```

Comment:
Looks like this function could return `Status` directly ?

Reply:
It's a little misunderstanding. However, Non-pipeline uses it as the follow codes. The return status `eval_const_conjuncts` indicates there is something wrong at evaluating, while the augment status is is guaranteed to be `OK` or `EndOfFile`.
```C++
Status OlapScanNode::open(RuntimeState* state) {
    SCOPED_TIMER(_runtime_profile->total_time_counter());
    RETURN_IF_CANCELLED(state);
    RETURN_IF_ERROR(ExecNode::open(state));

    Status status;
    RETURN_IF_ERROR(OlapScanConjunctsManager::eval_const_conjuncts(_conjunct_ctxs, &status));
    _update_status(status);

    _dict_optimize_parser.set_mutable_dict_maps(state, state->mutable_query_global_dict_map());
    DictOptimizeParser::rewrite_descriptor(state, _conjunct_ctxs, _olap_scan_node.dict_string_id_to_int_ids,
                                           &(_tuple_desc->decoded_slots()));

    return Status::OK();
}
```